### PR TITLE
#RI-1344 - Added tree view change delimiter

### PR DESCRIPTION
**Pull request by zalenskiSofteq**
#RI-1344 - Added tree view change delimiter

### CODE REVIEW

File: redisinsight/ui/src/components/virtual-tree/interfaces.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import { FixedSizeNodeData } from 'react-vtree'
 import { IKeyPropTypes } from 'uiSrc/constants/prop-types/keys'
 
-export const TREE_LEAF_FIELD = 'keys:keys'
+export const getTreeLeafField = (delimiter = '') => `keys${delimiter}keys`
```
Comment (romansergeenkosofteq) [inline redisinsight/ui/src/components/virtual-tree/interfaces.ts:4]: could you please move it from interfaces file?

File: redisinsight/ui/src/helpers/constructKeysToTree.ts [hunk 2]
```diff
@@ -2,21 +2,19 @@ import { IKeyPropTypes } from 'uiSrc/constants/prop-types/keys'
...
+  delimiter?: string
 }
 
 export const constructKeysToTree = (props: Props): any[] => {
-  const { items: keys, separator = ':' } = props
-  // const keysSymbol = Symbol(`keys${separator}keys`)
-  // const keysSymbol = Symbol('keys')
-  const keysSymbol = `keys${separator}keys`
+  const { items: keys, delimiter = ':' } = props
+  const keysSymbol = `keys${delimiter}keys`
```
Comment (romansergeenkosofteq) [inline redisinsight/ui/src/helpers/constructKeysToTree.ts:10]: maybe its better to use getTreeLeafField here

**Review by romansergeenkosofteq — DISMISSED** [review]

File: redisinsight/ui/src/components/virtual-tree/interfaces.ts (hunk 1 above)
Comment (zalenskiSofteq) [reply redisinsight/ui/src/components/virtual-tree/interfaces.ts]: moved

File: redisinsight/ui/src/helpers/constructKeysToTree.ts (hunk 2 above)
Comment (zalenskiSofteq) [reply redisinsight/ui/src/helpers/constructKeysToTree.ts:10]: This is file for WebWorker and I can't import getTreeLeafField function inside WebWorker

**Review by romansergeenkosofteq — APPROVED** [review]

**Comment by romansergeenkosofteq** [comment]
> @zalenskiSofteq @romansergeenkosofteq Where is the option in the UI (or other configuration) to change the delimiter from ":" to for example "::"?

Снимок экрана 2024-01-10 в 09 11 25

**Pull request merged**